crates/fluvio-sc/src/controllers/mirroring/controller.rs

```diff
@@ -0,0 +1,220 @@
+                    MIRRORING_CONTROLLER_RETRY_INTERVAL
+                );
+                sleep(Duration::from_secs(MIRRORING_CONTROLLER_RETRY_INTERVAL)).await;
+            }
+        }
+    }
+
+    #[instrument(skip(self))]
+    async fn inner_loop(&self) -> Result<()> {
+        debug!("initializing listeners");
+
+        loop {
+            let mirrors_from_home = self
+                .mirrors
+                .store()
+                .read()
+                .await
+                .values()
+                .filter_map(|r| match r.spec().mirror_type.clone() {
+                    MirrorType::Home(h) => Some(h),
+                    _ => None,
+                })
+                .collect::<Vec<_>>();
+
+            //TODO: we need to have a strategy to choose which home to connect
+            if let Some(mirror) = mirrors_from_home.first() {
+                //send to home cluster the connect request
+                //TODO: handle TLS
+
+                let home_config = ClientConfig::with_addr(mirror.public_endpoint.clone());
+                let versioned_socket = home_config.connect().await?;
+                let (socket, config, versions) = versioned_socket.split();
+                info!("connecting to home: {}", mirror.public_endpoint);
+
+                let request = MirrorConnect {
+                    name: mirror.remote_id.clone(),
+                };
+                debug!("sending connect request: {:#?}", request);
+
+                let mut stream_socket = fluvio::stream_socket::StreamSocket::new(
+                    config,
+                    MultiplexerSocket::shared(socket),
+                    versions.clone(),
+                );
+
+                let version = versions
+                    .lookup_version::<ObjectMirroringRequest>()
+                    .ok_or(anyhow!("no version found for mirroring request"))?;
+
+                let req = ObjectMirroringRequest::try_encode_from(
+                    MirroringRemoteClusterRequest { request },
+                    version,
+                )?;
+
+                let mut stream = stream_socket
+                    .create_stream_with_version(req, version)
+                    .await?;
+
+                while let Some(response) = stream.next().await {
+                    // Change status to connected
+                    let now = SystemTime::now()
+                        .duration_since(SystemTime::UNIX_EPOCH)?
+                        .as_millis();
+                    match response {
+                        Ok(response) => {
+                            debug!("received response: {:#?}", response);
+                            for topic in response.topics.iter() {
+                                // Check if the topic already exists
+                                if self.topics.store().read().await.contains_key(&topic.key) {
+                                    debug!("topic {} already exists", topic.key);
```

Comment:
also need to check configuration is same.  If configuration is different then need to apply diff logic

Reply:
I added a new diff logic on last commit